dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/utils/GithubAppCheck.java

```diff
@@ -76,11 +76,13 @@ public GithubAppCheck(final Namespace ns){
      */
     protected boolean isGithubAppEnabledOnRepository(String fullRepoName) {
         Integer maxRetryCount = 5;
-        Boolean isGithubAppInstalled = isGithubAppEnabledOnRepositoryWithGitApi(fullRepoName);
+        Boolean isGithubAppInstalled = false;
         if (!isGithubAppInstalled) {
             for (Integer i = 0; i < maxRetryCount; i++) {
```

Comment:
Did we explore @jeetchoudhary 's suggestion of using retry handlers provided by HTTP? Or Resilience4J is a good library for retries with exponential backoff.

Reply:
```
Sure, here's how you can refactor the `isGithubAppEnabledOnRepository` and `isGithubAppEnabledOnRepositoryWithGitApi` methods to use the Resilience4j library for implementing backoff and retries for requests to GitHub.

Resilience4j is a lightweight fault tolerance library inspired by Netflix Hystrix, but designed for Java 8 and functional programming. It provides several modules such as Retry, CircuitBreaker, RateLimiter, and TimeLimiter, which can be used standalone or combined with each other.

```java
import io.github.resilience4j.retry.Retry;
import io.github.resilience4j.retry.RetryConfig;
import io.vavr.control.Try;

import java.time.Duration;

// ...

public class GithubAppCheck {
    // ...

    /**
     * Method to verify whether the github app is installed on a repository or not. 
     * @param fullRepoName = The repository full name, i.e, of the format "owner/repoName". Eg: "Salesforce/dockerfile-image-update"
     * @return True if github app is installed, false otherwise. 
     */
    protected boolean isGithubAppEnabledOnRepository(String fullRepoName) {
        RetryConfig config = RetryConfig.custom()
                .maxAttempts(5)
                .waitDuration(Duration.ofMillis(1000))
                .retryExceptions(IOException.class)
                .build();
        Retry retry = Retry.of("id", config);

        Try<Boolean> retryResult = Try.ofSupplier(Retry.decorateSupplier(retry, () -> isGithubAppEnabledOnRepositoryWithGitApi(fullRepoName)))
                .recover(throwable -> isGithubAppEnabledOnRepositoryWithRenovateApi(fullRepoName));

        return retryResult.get();
    }

    /**
     * Method to verify whether the github app is installed on a repository or not, using Git API
     * @param fullRepoName = The repository full name, i.e, of the format "owner/repoName". Eg: "Salesforce/dockerfile-image-update"
     * @return True if github app is installed, false otherwise. 
     * Reference: https://docs.github.com/en/rest/apps/apps?apiVersion=2022-11-28#get-a-repository-installation-for-the-authenticated-app
     */
    protected boolean isGithubAppEnabledOnRepositoryWithGitApi(String fullRepoName) {
        refreshJwtIfNeeded(appId, privateKeyPath);
        try {
            gitHub.getApp().getInstallationByRepository(fullRepoName.split("/")[0], fullRepoName.split("/")[1]);
            return true;
        } catch (HttpException exception) {
            if (exception.getResponseCode() != 404) {
                // Log for any HTTP status code other than 404 Not found. 
                log.warn("Caught a HTTPException `{}` while trying to get app installation. Defaulting to False. Status code: {}", exception.getMessage(), exception.getResponseCode());
            }
            throw exception;
        } catch (IOException exception) {
            // Most often happens on timeout scenarios. 
            log.warn("Caught a IOException {} while trying to get app installation. Defaulting to False", exception.getMessage());
            throw exception;
        }
    }
    // ...
}
```
In the above code, a `Retry` instance is created with a configuration that specifies the maximum number of retry attempts and the wait duration between each attempt. The `retryExceptions` method is used to specify the types of exceptions that should trigger a retry. The `Retry.decorateSupplier` method is used to create a retryable version of the `isGithubAppEnabledOnRepositoryWithGitApi` method. If the method throws an exception, the `Retry` instance will automatically retry the method according to the specified configuration. If all retry attempts fail, the `recover` method is used to fall back to the `isGithubAppEnabledOnRepositoryWithRenovateApi` method.
```

Reply:
@cjneasbi This has been applied